autoPyTorch/api/base_task.py

```diff
@@ -1,139 +1,740 @@
+        self._stopwatch = StopWatch()
 
-        self.pipeline_config = self.pipeline.get_default_config()
+        self.default_pipeline_options = replace_string_bool_to_bool(json.load(open(
+            os.path.join(os.path.dirname(__file__), 'default_pipeline_options.json'))))
 
-        self.search_space = self.pipeline.get_hyperparameter_search_space()
+        self.search_space: Optional[ConfigurationSpace] = None
+        self._dataset_requirements: Optional[List[FitRequirement]] = None
+        self.task_type: Optional[str] = None
+        self._metric: Optional[autoPyTorchMetric] = None
+        self._logger: Optional[PicklableClientLogger] = None
+        self.run_history: Optional[RunHistory] = None
+        self.trajectory: Optional[List] = None
+        self.dataset_name: Optional[str] = None
+        self.cv_models_: Dict = {}
+
+        # By default try to use the TCP logging port or get a new port
+        self._logger_port = logging.handlers.DEFAULT_TCP_LOGGING_PORT
+
+        # Store the resampling strategy from the dataset, to load models as needed
+        self.resampling_strategy = None  # type: Optional[Union[CrossValTypes, HoldoutValTypes]]
 
     @abstractmethod
-    def build_pipeline(self):
+    def _get_required_dataset_properties(self, dataset: BaseDataset) -> Dict[str, Any]:
+        """
+        using the dataset entered, this function will
+        return the required dataset properties given the task
+        """
         raise NotImplementedError
 
     def set_pipeline_config(
             self,
             **pipeline_config_kwargs: Any) -> None:
         """
-        Check wether arguments are valid and then pipeline configuration.
+        Check whether arguments are valid and
+        then sets them to the current pipeline
+        configuration.
+        Args:
+            **pipeline_config_kwargs: Valid config options include "job_id",
+            "device", "budget_type", "epochs", "runtime", "torch_num_threads",
+            "early_stopping", "use_tensorboard_logger", "use_pynisher",
+            "metrics_during_training"
+
+        Returns:
+            None
         """
         unknown_keys = []
-        for option, value in pipeline_config_kwargs.items:
-            if option in self.pipeline_config.keys():
+        for option, value in pipeline_config_kwargs.items():
+            if option in self.default_pipeline_options.keys():
                 pass
             else:
                 unknown_keys.append(option)
 
         if len(unknown_keys) > 0:
-            raise ValueError("Invalid configuration arguments given %s" % unknown_keys)
+            raise ValueError("Invalid configuration arguments given {},"
+                             " expected arguments to be in {}".
+                             format(unknown_keys, self.default_pipeline_options.keys()))
 
-        self.pipeline_config.update(pipeline_config_kwargs)
+        self.default_pipeline_options.update(pipeline_config_kwargs)
```

Comment:
Probably should rename `self.default_pipelie_options` to `self.pipeline_options` since this is not the default anymore?